### DISCUSSION TITLE
Using mitmproxy in a production environment

### DISCUSSION THREAD

**Discussion by odehneh** [mitmproxy/mitmproxy#6813] [Q&A]
Hello all
I'm planning to use mitmproxy to transparently downgrade TLS traffic version between internal clients and external servers, specifically by using the tls_version_server_max options. The objective is to enable the use of a URL filtering products that only support TLS 1.2. Of course I will add the mitmproxy certificates as a trusted toot certificate on clients.

client----(TLS1.3)-------mitmproxy------(TLS1.2)-------URLF-------Internet

I have 2 questions
1) is the above possible?
2) Can it be used in a production environment?  (~50 Mbps, 75 users)

**Comment by emanuele-em** [comment]
1. mitmrpoxy should do that by default: [source](https://stackoverflow.com/questions/69938653/q-mitmproxy-negotiate-tls-downgrade-on-internal-nic)
2. is not meant to be used in place of ngix or others, and its speed is not comparable, you risk it becoming your bottleneck, you should try to test before
